drivers/clock_control/clock_control_nrfs_audiopll.c

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrfs_audiopll
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/nrf_clock_control.h>
+
+#include <nrfs_audiopll.h>
+#include <nrfs_backend_ipc_service.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(nrfs_audiopll, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#define SHIM_MIN_FREQUENCY 10666667
+#define SHIM_MAX_FREQUENCY 13333293
+
+#define SHIM_DEFAULT_PRESCALER AUDIOPLL_DIV_4
+
+BUILD_ASSERT(
+	DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	"multiple instances not supported"
+);
+
+BUILD_ASSERT(DT_INST_PROP(0, frequency) >= SHIM_MIN_FREQUENCY);
+BUILD_ASSERT(DT_INST_PROP(0, frequency) <= SHIM_MAX_FREQUENCY);
+
+struct shim_data {
+	const struct device *dev;
+	uint16_t freq_fraction;
+	struct k_sem evt_sem;
+	nrfs_audiopll_evt_type_t evt;
+};
+
+struct shim_config {
+	uint32_t frequency;
+};
+
+static void shim_nrfs_audiopll_evt_handler(nrfs_audiopll_evt_t const *evt, void *context)
+{
+	struct shim_data *dev_data = context;
+
+	dev_data->evt = evt->type;
+	k_sem_give(&dev_data->evt_sem);
+}
+
+static int shim_nrfs_request_enable(const struct device *dev)
+{
+	struct shim_data *dev_data = dev->data;
+	nrfs_err_t err;
+
+	LOG_DBG("send enable request");
+
+	err = nrfs_audiopll_enable_request(dev_data);
```

Comment:
not sure if there is a case for that but API is not thread safe. `dev_data->evt` may get overwritten if two calls are called one after another (from two contexts) and both response handlers are received before first context processes `dev_data->evt`. Maybe a simple atomic flag that returns `-EBUSY` if api is called when there is already active call?

Reply:
The driver is not thread safe by design, thread safety is not a requirement of drivers which are not meant to be shared, like busses for example :)